src/language-js/postprocess.js

```diff
@@ -2,17 +2,21 @@
 
 const { getLast } = require("../common/util");
 
-// fix unexpected locEnd caused by --no-semi style
 function postprocess(ast, options) {
   visitNode(ast, node => {
     switch (node.type) {
+      // Fix unexpected locEnd caused by --no-semi style
       case "VariableDeclaration": {
         const lastDeclaration = getLast(node.declarations);
         if (lastDeclaration && lastDeclaration.init) {
           overrideLocEnd(node, lastDeclaration);
         }
         break;
       }
+      case "TSAsExpression": {
+        convertTSAsExpression(node);
```

Comment:
Feels pretty hacky to me, but I'm not sure how to do it better. There's quite a bit of code for printing binary expressions. On the other hand – do `as` expressions really need to go through _all_ of the binary expression stuff, like the JSX related parts? Maybe there’s a way to extract or copy the parts needed for `as`?

@vjeux You have had useful insight before on PRs with hacky code – what does your experience tell you here (gut feeling)?

Reply:
> Maybe there’s a way to extract or copy the parts needed for as?

My thought here was that keeping the two in sync would be a bigger maintenance burden, but if that's the preferred way of doing things, I can definitely look into it!

I'm also coming from working on babel-eslint, where the parser has to do a lot of massaging of the AST to get it to work with ESLint.

Reply:
Does it modify the passed AST itself or a copy of it? What if someone else wanted to use that AST after Prettier?

Reply:
> do as expressions really need to go through all of the binary expression stuff, like the JSX related parts

Why not? `as` can be used anywhere, just like `+` or say `&&`,

Reply:
Happy to work on this again once the team decides what the preferred approach is!